File: src/main/java/de/tectoast/emolga/commands/pokemon/RandomizeKillsCommand.java
package de.tectoast.emolga.commands.pokemon;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;

import java.util.Collections;
import java.util.LinkedList;
import java.util.List;
import java.util.Random;

public class RandomizeKillsCommand extends Command {

    public RandomizeKillsCommand() {
        super("randomizekills", "Randomized die Kills auf 6 Mons", CommandCategory.Pokemon);
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) throws Exception {
        List<Integer> l = new LinkedList<>();
        Random r = new Random();
        for (int i = 0; i < 6; i++) {
            int rand = r.nextInt(6) + 1;
            int sum = l.stream().mapToInt(x -> x).sum();
            if (sum + rand > 6) rand = 6 - sum;
            l.add(rand);
        }
        Collections.shuffle(l);
        StringBuilder b = new StringBuilder();
        for (int i = 0; i < l.size(); i++) {
            b.append("Pokemon ").append(i + 1).append(": ").append(l.get(i)).append("\n");
        }
        e.reply(b.toString());
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/pokemon/RandomizeKillsCommand.kt
package de.tectoast.emolga.commands.pokemon

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import java.util.*

class RandomizeKillsCommand : Command("randomizekills", "Randomized die Kills auf 6 Mons", CommandCategory.Pokemon) {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val l: MutableList<Int> = LinkedList()
        val r = Random()
        for (i in 0..5) {
            var rand = r.nextInt(6) + 1
            val sum = l.sum()
            if (sum + rand > 6) rand = 6 - sum
            l.add(rand)
        }
        l.shuffle()
        e.reply(l.indices.joinToString("\n") { "Pokemon ${it + 1}: ${l[it]}" })
    }
}